include/nova_renderer/frontend/procedural_mesh.hpp

```diff
@@ -2,8 +2,13 @@
 
 #include <array>
 #include <cstdint>
+#include <optional>
 
 #include "nova_renderer/constants.hpp"
+#include "nova_renderer/rhi/forward_decls.hpp"
+
+#include "../../../src/memory/block_allocation_strategy.hpp"
```

Comment:
This include is a problem. It is referencing a private header from the public API

Reply:
Fixed